src/main/java/org/broadinstitute/consent/http/db/MatchDAO.java

```diff
@@ -30,8 +30,17 @@ public interface MatchDAO extends Transactional<MatchDAO> {
     @SqlQuery("SELECT * FROM match_entity WHERE matchid = :id ")
     Match  findMatchById(@Bind("id") Integer id);
 
-    @SqlQuery("SELECT * FROM match_entity WHERE purpose IN (<purposeId>)")
-    List<Match> findMatchesForPurposeIds(@BindList("purposeId") List<String> purposeId);
+    @SqlQuery(
+        "SELECT match_entity.* FROM match_entity " +
+        "INNER JOIN (" +
+        "   SELECT election.*, MAX(election.electionid) OVER (PARTITION BY election.referenceid, election.datasetid) AS latest " +
+        "   FROM election " +
+        "   WHERE LOWER(election.electiontype) = 'dataaccess' " +
+        ") AS e " +
+        "ON e.referenceid = match_entity.purpose " +
+        "WHERE match_entity.purpose IN (<purposeIds>) AND e.electionid = latest"
+    )
+    List<Match> findMatchesForPurposeIds(@BindList("purposeIds") List<String> purposeIds);
```

Comment:
That's not true. It's the reason why Shae had to use id for a foreign key instead of reference id for [DUOS-1519](https://github.com/DataBiosphere/consent/pull/1551). Reference ids are supposed to be unique, but considering that staging has duplicates the possibility exists for dev and prod as well.

Reply:
> That's not true. It's the reason why Shae had to use id for a foreign key instead of reference id for https://github.com/DataBiosphere/consent/pull/1551. Reference ids are supposed to be unique, but considering that staging has DARs with the same reference id we can assume the possibility exists for dev and prod as well.

I wasn't aware of that. If that is true, it's a very serious bug. They HAVE to be unique so elections can refer to a single DAR and not multiple DARs. We need to file a ticket for this if true.